### ISSUE TITLE
RE Code Veronica X crashes as I enter Floor 1F with Chris

### ISSUE THREAD

**Issue by jclsn** [libretro/flycast#784]
I am on a Raspberry Pi 4 with Lakka 2.3.1 and I can't continue the game, because it crashes when I select floor 1F shortly after the episode with Chris Redfield started. Tried to reextract and copy the rom again, but it doesn't help. Before the game worked really well!

**Comment by flyinghead** [comment]
Can you upload a VMU file or savestate to help reproduce the issue?

**Comment by jclsn** [comment]
This save state is inside the elevator http://www.mediafire.com/file/ftl7vr0642v73cf/Resident_Evil_-_Code_Veronica_X_v1.000_%25282000%2529%2528Capcom%2529%2528NTSC%2529%2528US%2529%2528Disc_2_of_2%2529%255B%2521%255D.state1/file

**Comment by flyinghead** [comment]
I'm getting to an error screen that reads `RDT version error` regardless of the floor I choose when loading your state. Is it what you're seeing on your end?

**Comment by jclsn** [comment]
I don't see any error message. As soon as the door of the elevator opens, Lakka reboots

**Comment by flyinghead** [comment]
Do you also have a VMU that I could use? I want to rule out any issue with the save state itself if possible.

**Comment by jclsn** [comment]
Here is my first VMU, that should contain the savestate. State 12 was the last one I made before I made the Retroarch state. State 13 is a state I made after loading the Retroarch state, so if the state was corrupted this one might be too. I tried loading state 13 and going back to the elevator, but the game also crashes. State 12 still needs some way to go, until you can enter the elevator.

https://www.mediafire.com/file/5i3wabzemr2fjln/vmu_save_A1.bin/file

**Comment by flyinghead** [comment]
I don't see any state on this vmu. It shows all 15 slots as "no data"

**Comment by jclsn** [comment]
It is Code Veronica X. Do you maybe use the normal Code Veronica? 






On Sun, Jan 19, 2020 at 10:19 PM +0100, "flyinghead" <notifications@github.com> wrote:












I don't see any state on this vmu. It shows all 15 slots as "no data"

**Comment by flyinghead** [comment]
Indeed I was using the regular version, not the X one. That explains why the savestate was crashing and why the VMU seemed empty. Now I'm back to your initial savestate, and just reproduced the crash when reaching floor 1F.

**Comment by jclsn** [comment]
Great! Hope you can fix it




On Mon, Jan 20, 2020 at 6:11 PM +0100, "flyinghead" <notifications@github.com> wrote:












Indeed I was using the regular version, not the X one. That explains why the savestate was crashing and why the VMU seemed empty. Now I'm back to your initial savestate, and just reproduced the crash when reaching floor 1F.

**Comment by DukeSkinny** [comment]
@coldspark29 Is this crash still occurring?

**Comment by jclsn** [comment]
Unfortunately I lost my savegame. Do you still have the VMU file?

**Comment by flyinghead** [comment]
Here is your VMU:
[vmu_save-RE2-floor1f-crash.zip](https://github.com/libretro/flycast/files/5585968/vmu_save-RE2-floor1f-crash.zip)

I recently tried the savestate and it's still crashing

**Comment by jclsn** [comment]
Thanks. Good to have that back!. It does still crash for me, but I am also using Lakka. So Flycast might not be fully up to date. When did you fix it?

**Comment by jclsn** [comment]
Yes, it still segfaults, but the lighting has seemed to have improved :)

```
[libretro ERROR] Verify Failed  : full_table_size < ARRAY_SIZE(full_table)
 in cache_entry -> core/hw/sh4/modules/fastmmu.cpp : 59 

[libretro ERROR] Verify Failed  : full_table_size < ARRAY_SIZE(full_table)
 in cache_entry -> core/hw/sh4/modules/fastmmu.cpp : 59 

[libretro ERROR] Verify Failed  : full_table_size < ARRAY_SIZE(full_table)
 in cache_entry -> core/hw/sh4/modules/fastmmu.cpp : 59 

[libretro ERROR] Verify Failed  : full_table_size < ARRAY_SIZE(full_table)
 in cache_entry -> core/hw/sh4/modules/fastmmu.cpp : 59 

Segmentation fault
```

**Comment by flyinghead** [comment]
Fixed in standalone by this commit https://github.com/flyinghead/flycast/commit/b3de6a166df68e972d3bed44d25806123b773673

**Comment by jclsn** [comment]
Nice! Thx
